server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -171,199 +162,105 @@ static void clearLoadedModules()
       }
     }
 
-    if (config.searchCurrentClassloader()) {
-      for (T module : ServiceLoader.load(clazz, Initialization.class.getClassLoader())) {
-        String moduleName = module.getClass().getCanonicalName();
-        if (moduleName == null) {
-          log.warn(
-              "Extension module [%s] was ignored because it doesn't have a canonical name, is it a local or anonymous class?",
-              module.getClass().getName()
-          );
-        } else if (!extensionNames.contains(moduleName)) {
-          log.info("Adding local extension module[%s] for class[%s]", moduleName, clazz.getName());
-          extensionNames.add(moduleName);
-          retVal.add(module);
-        }
-      }
-    }
-
     // update the map with currently loaded modules
     extensionsMap.put(clazz, retVal);
 
     return retVal;
   }
 
-  public static URLClassLoader getClassLoaderForCoordinates(
-      TeslaAether aether,
-      String coordinate,
-      String defaultVersion
-  )
-      throws DependencyResolutionException, MalformedURLException
+  /**
+   * Find all the extension files that should be loaded by druid.
+   * <p/>
+   * If user explicitly specifies druid.extension.toLoad, then it will look for those extensions under root
+   * extensions directory. If one of them is not found, druid will fail loudly.
+   * <p/>
+   * If user doesn't specify druid.extension.toLoad (or its value is empty), druid will load all the extensions
+   * under the root extensions directory.
+   *
+   * @param config ExtensionsConfig configured by druid.extensions.xxx
+   *
+   * @return an array of druid extension files that will be loaded by druid process
+   */
+  public static File[] getExtensionFilesToLoad(ExtensionsConfig config)
```

Comment:
How pedantic do we want to be here?

Should we be checking that not only these files exist but that they are directories?